src/Reflection/ReflectionClass.php

```diff
@@ -1238,4 +1244,17 @@ public function removeProperty($propertyName)
 
         return false;
     }
+
+    private function findTypeFromAst($namespace, $locatedSource, $type)
```

Comment:
> Wouldn't it make more sense to create the context in FindTypeForAst helper itself

Well, seems unnecessary, as we would just be injecting the factory which would always be invoked in the same way.

> Or, inject the context into the FindType* classes constructor

This is what happens in this PR.

>  I'd prefer to not have them all bunched up into one class

I don't think its particularly elegant either, will search for another solution